sklearn/preprocessing/resample.py

```diff
@@ -0,0 +1,263 @@
+            return scaling
+        else:
+            raise ValueError("Invalid value for scaling, must be "
+                             "float, int, or None: %s" % scaling)
+
+
+def _weighted_sample(probas, n_samples, random_state=None):
+    """Select indices from n_samples with a weighted probability.
+
+    Parameters
+    ---------
+    probas : array-like
+        Array of probabilities summing to 1.
+    n_samples : integer
+        The number of samples to draw from at random.
+    random_state : int, or RandomState instance (optional)
+        Control the sampling for reproducible behavior.
+    """
+    random_state = check_random_state(random_state)
+    if abs(np.sum(probas) - 1.0) > .011:
+        raise ValueError("Label distribution probabilites must sum to 1")
+    cum_probas = np.cumsum(probas)
+    cum_probas[-1] = 1  # ensure that the probabilities sum to 1
+    space = np.linspace(0, 1, 10000)
+    weighted_indices = np.searchsorted(cum_probas, space)
+    rs = random_state.randint(0, high=len(space), size=n_samples)
+    return weighted_indices[rs]
+
+
+def resample_labels(y, method=None, scaling=None, replace=False,
+                    shuffle=False, random_state=None):
+    """Resamples a classes array `y` and returns an array of indices
+
+    The default behavior is to output the same ``y``. The additional
+    parameters control the desired class distribution of the indices and the
+    number of samples in the output.
+
+    Parameters
+    ----------
+    y : array-like, shape (n_samples,)
+        Target classes. Pass in the entire classes array so that this
+        function can work on the class distribution.
+    method : "balance", "oversample", "undersample", dict (optional)
+        None outputs samples with the same class distribution as `y`.
+        "balance" rebalances the classes to be equally distributed,
+            over/undersampling for `len(y)` samples by default.
+        "oversample" grows all classes to the count of the largest class.
+        "undersample" shrinks all classes to the count of the smallest class.
+        dict with pairs of class, probability with values summing to 1
+    scaling : integer, float (optional)
+        Number of samples to return.
+        None outputs the same number of samples.
+        `integer` is an absolute number of samples.
+        `float` is a scale factor.
+    replace : boolean (False by default)
+        Sample with replacement when True.
+    shuffle : boolean (False by default)
+        Shuffle the indices before returning them. This option can add
+        significant overhead, so it is disabled by default.
+    random_state : int, or RandomState instance (optional)
+        Control the sampling for reproducible behavior.
+
+    Returns
+    -------
+    indices : array-like, shape (n_samples,)
+        Indices sampled from the dataset respecting a class distribution
+        controlled by this function's parameters.
+
+    Examples
```

Comment:
This seems to me a lot of examples, with too much changing between them (i.e. `X` and `y` could be constant). Could this be simplified?